pandas/_typing.py

```diff
@@ -40,7 +40,9 @@
 # array-like
 
 AnyArrayLike = TypeVar("AnyArrayLike", "ExtensionArray", "Index", "Series", np.ndarray)
+AnyArrayLikeUnion = Union["ExtensionArray", "Index", "Series", np.ndarray]
```

Comment:
can you add a comment on when you should use the AnyArryLike TypeVar and when the Union (for readers of this code!)